backend/dal/Helpers/Extensions/LeaseExtensions.cs

```diff
@@ -140,6 +144,25 @@ private static IQueryable<Entities.PimsLease> GenerateCommonLeaseQuery(this IQue
             return query;
         }
 
+        /// <summary>
+        /// Checks for given source field name and if found replaces it with target field name for sorting the data in given sort array
+        /// </summary>
+        /// <param name="sourceField">Sort field name from model</param>
+        /// <param name="targetField">Sort field name from entity</param>
+        /// <param name="sortDef">Find and replaces the soft field in this array</param>
+        private static void MapSortField(string sourceField, string targetField, string[] sortDef)
+        {
+            var sortField = sortDef.FirstOrDefault(x => x.Contains(sourceField));
+            if (sortField != null)
+            {
+                var sortFieldIndex = sortDef.ToList().IndexOf(sortField);
+                if (sortFieldIndex > -1)
```

Comment:
I think I'm in favour of throwing an error here, as this not having a matching sort field is likely a development-time error which we would want to fail fast on.

Reply:
That makes sense. I am thinking we could improve the sorting + filters to avoid having possible runtime errors